### PR TITLE
add values to charter

### LINKED ISSUE #102: Adding group Values
We should add our group values to make it clear.

Values / Principles
It is also good practice to include a statement of values or principles within your governance documentation. While the scope includes information about what your project does, your values / principles define how you work. They often include statements about openness, transparency, inclusion, being welcoming and respectful, and much more.

These are all living documents that should be expected to change over time as the project evolves. For sandbox projects, this might be a simple one or two sentence statement about what the project does, and by the time a project has graduated, they would probably have a more detailed mission statement, scope, and values / principles. All of this documentation should be consistent with the mission and values in the CNCF Charter.

https://docs.google.com/document/d/1QawLfsz2_n2Y0nPuMzTy9tF9Tu3iFanq5PJgXK-yr3Y/edit?ts=604bb3e4

### PR DESCRIPTION
**Pull request by lixuna**
Refs #102 

wording from: 
- https://github.com/vitessio/vitess/blob/master/GUIDING_PRINCIPLES.md, 
- https://github.com/goharbor/community/blob/master/GUIDING_PRINCIPLES.md,
- https://github.com/tikv/community/blob/master/guiding-principles.md

### CODE REVIEW

**Review by taylor — APPROVED** [review]

File: charter.md [hunk 1]
```diff
@@ -9,6 +9,15 @@ The goals for the group are:
...
 
+## Values
+The CNF WG is driven by high technical standards, and these must be maintained. It is also important to take into account that multiple humans are participating in the project, and we must ensure that we all treat each other well. In this respect, we will honor the following values:
+* Technical excellence
+* Innovation and creativity
+* Fairness and equality
+* Diversity
+* Inclusiveness
+* Openness
+* Adherence to the [Code of Conduct](code-of-conduct.md)
```
Comment (xmulligan) [inline charter.md:20]:
```suggestion
The CNF WG is driven by high technical standards, and these must be maintained. It is also important to take into account that multiple humans are participating in the project, and we must ensure that we all treat each other well. In this respect, we will honor the following values:

* Technical excellence over duct tape
Networking technologies power the underlying fabrics of our lives, including life critical applications, some of which runs on decades old technology. The work we produce could be affecting lives decades from now. Our technical quality should match these outcomes.

* Inclusive is better than exclusive
Broadly successful and useful technologies require different perspectives and skill sets, which can only be heard in a welcoming and respectful environment. Community membership is a privilege, not a right. Community members earn leadership through effort, scope, quality, quantity, and duration of contributions to the technology and the community.

* Evolution is better than stagnation
The CNG WG is not meant to be a set in time standard. The best practices should continually evolve with state of the art in the Kubernetes and cloud native ecosystems. Community leaders also have a duty to find, sponsor, and promote new community members. Leaders should expect to step aside. Community members should expect to step up.

* Adherence to the [Code of Conduct](code-of-conduct.md)
```

Comment (taylor) [reply charter.md:20]: Diversity should be mentioned as a goal for having using inclusive methods + process.  Diversity is what brings different perspectives, skill sets, and ideas. Inclusiveness is a way to get to the goal of diversity.

Evolution is about change, Innovation is about introducing new ideas, techniques and processes. 

- Technical excellence
- Evolution, Innovation, and creativity
- Diversity through openness and inclusiveness
- Adherence to the CNCF Code of Conduct

Comment (xmulligan) [reply charter.md:20]: @taylor please add something to the "* Inclusive is better than exclusive" section if you feel something is missing

File: charter.md [hunk 2]
```diff
@@ -9,6 +9,15 @@ The goals for the group are:
...
```
Comment (xmulligan) [inline charter.md:21]:
```suggestion
* Commit first, improve later
The community thrives on contribution. A PR or suggestion for a change will move the community forward faster. Healthy disagreements are important for technical excellence, but should be accompanied with a way to move forward. We will accept good contributions and continue to make them better later

* Pragmatism over Politics
Our work should be focused on what works best in the real world rather than what works best for one group. Everything we do should have a basis in pratical applications and use cases.
```

Comment (iawells) [reply charter.md:21]: Alliteration over ... ? :)

I think this works (the first point loosely covers 'commit first, fix later' as well)

File: charter.md (hunk 1 above)
Comment (iawells) [reply charter.md:20]: I think inclusive/exclusive says that you're judged on output and not collaboration and actually undermines its headline (see also: RMS). Can we say that you are judged on both output and also on engaging others within the community?

File: charter.md (hunk 2 above)
Comment (taylor) [reply charter.md:21]: @iawells I agree with the commit first, fix later... or commit often and expect failure... 

I'm trying to think of something other than criticism though.  Tests are criticism which are evaluating and judging which is part of the definition of criticism. That is a contribution.

@xmulligan I agree with providing a way to move forward when possible. In the case of tests it may be that a fault can be detected but communicating a correct way forward may be difficult or faulty (lack of context, many paths, etc).

I think what you are suggestion is something like "Solutions over complaints".  I'll reiterate a criticism very well can be a contribution.

@iawells,The commit early and often can be its own value. I'm 👍 on it.

File: charter.md (hunk 1 above)
Comment (taylor) [reply charter.md:20]: @iawells I'm not clear on what you are saying.  

To be clear on my views on output, behaviour, inclusive vs diversity:
- Great output does not compensate for toxic behaviour --> I prefer not having a "toxic genius"
- Inclusive and diverse are not the same. diversity is a goal and brings results usually mentioned for inclusiveness are actually about diversity.

File: charter.md (hunk 2 above)
Comment (iawells) [reply charter.md:21]: Aphorisms would include 'don't let the perfect be the enemy of the good' - meaning: if you think something is good but could be better, then accept the good and then make it better afterward.

File: charter.md (hunk 1 above)
Comment (iawells) [reply charter.md:20]: "Inclusive is better than exclusive
[...]
Community membership is a privilege, not a right. Community members earn leadership through effort, scope, quality, quantity, and duration of contributions"

I intended to refer to it as being in the "inclusive/exclusive" section when I mentioned that; not that it was the only criterion for inclusiveness.  I could have used my words...

That is the point that I was saying looked output and not collab-focussed (and perhaps because I read 'contribution' as 'commit', when that may not have been your intention).

I read this as: we strive to value people for what they produce (and, with collab included, how they encourage others - in summary, the good that they do); and because we appreciate the need for a wide assortment of views we will uphold standards that lead to a welcoming environment.

File: charter.md (hunk 2 above)
Comment (taylor) [reply charter.md:21]: I'm hearing echos of "worse is better".  Are we headed to the New Jersey style vs MIT/Standford style debate --> Worse is Better vs the Right Thing? :)

Comment (taylor) [reply charter.md:21]: @iawells i think a value that communicates early contributions over perfection would be good.  Up for writing that?

**Comment by taylor** [comment]
@xmulligan why not go with a shorter word list of values before expanding?

After going back and reviewing the original PR vs what we are discussing now it seems we are moving towards perfection before early contribution.  We are more likely to have problems the more specific we get so we have to take longer to get the wording right.

**Comment by taylor** [comment]
An alternative approach to adding our CNF WG values is to request our community to submit values it believes in then we select and vote in adoption of them as a community.

File: charter.md (hunk 2 above)
Comment (jeffsaelens) [reply charter.md:21]: Maybe instead just go with something like "Continuous Feedback and/or Iteration"? If you want to get rid of criticism then maybe rephrase the heading so that it promotes the idea that this will be a series of endless contributions that slowly build to something better.

Comment (jeffsaelens) [reply charter.md:21]: "rather than what works best for one vendor" I don't think we should single out vendors. OpenSource devs want to see their creations rise to the top. Providers want things their way and bigger ones are fans of pushing their agendas on smaller ones. 

I know I am often guilty of picking on vendors trying to sell their solutions through opensource infiltration, but I'm also aware of the warts my side has as welkl.

File: charter.md (hunk 1 above)
Comment (jeffsaelens) [reply charter.md:20]: maybe add to the end of the inclusive section, ... of contributions "to the technology and community"? This sort of touches on the point that it's not just about uncovering the best practices, but fostering that which makes opensource such a powerhouse, to begin with.

**Review by jeffsaelens — COMMENTED** [review]
posted some initial thoughts. 

edit: none of which ended up in the right place apparently...

File: charter.md (hunk 1 above)
Comment (xmulligan) [reply charter.md:20]: I didn't intend it to be all about technical contributions to the detriment of everything else. I've added @jeffsaelens 's suggestion about tech + community. If anyone else has a better way to word it, I would be happy to add it.

File: charter.md (hunk 2 above)
Comment (xmulligan) [reply charter.md:21]: I would be happy to change the first title from "contribution over criticism" to "Commit first, fix later" and adding "We will accept good contributions and continue to make them better later"

for the second, what about changing it from "vendor" to "group"?

Comment (taylor) [reply charter.md:21]: @xmulligan 
+1 on commit first, fix later type wording.  Maybe something like "Commit early, improve later?" vs fix?

+1 for group replacing vendor

Comment (iawells) [reply charter.md:21]: If it helps, one phrase that I heard years ago is 'the egoless programmer' - the one that wants to do the right thing more than their own thing.  I wonder if we could use something like that over the whole vendor phrasing - just a thought in case it turns out to be a more useful tack.

Comment (xmulligan) [reply charter.md:21]: I've updated both using the suggestions above. @lixuna would you be willing to commit this to your PR?

File: charter.md (hunk 1 above)
Comment (xmulligan) [reply charter.md:20]: Any more comments? @lixuna would you add this suggestion?

File: charter.md (hunk 2 above)
Comment (taylor) [reply charter.md:21]: It seems the alliteration version is based on what K8s uses https://github.com/kubernetes/community/blob/master/values.md

File: charter.md (hunk 1 above)
Comment (taylor) [reply charter.md:20]: Technical high-standards and shorter list is partly from Vitess values https://github.com/vitessio/vitess/blob/master/GUIDING_PRINCIPLES.md#values.  These are also referenced in the this [PR description](https://github.com/cncf/cnf-wg/pull/119#issue-604145088) also.

Comment (taylor) [reply charter.md:20]: @xmulligan can we drop the technical excellence item and just keep the original vitess starter text?

> * Technical excellence over duct tape
> Networking technologies power the underlying fabrics of our lives, including life critical applications, some of which runs on decades old technology. The work we produce could be affecting lives decades from now. Our technical quality should match these outcomes.

We will move towards better as communicated in the "commit early and improve later"

**Comment by taylor** [comment]
The new changes are a full overhaul.  I suggest we move them to a new PR and accept this PR as-is.  I've created a PR which does that https://github.com/cncf/cnf-wg/pull/134, copying all of the current suggested commits there. (note: it can be updated to go against `main` if this PR is accepted)

@xmulligan @iawells @jeffsaelens @lixuna 

If you agree, please approve the PR

**Comment by taylor** [comment]
cc: @electrocucaracha @claudiobartolini @vukg re: https://github.com/cncf/cnf-wg/pull/119#issuecomment-819695297

If you agree, please approve the PR

File: charter.md (hunk 2 above)
Comment (taylor) [reply charter.md:21]: These changes are moved to the new PR https://github.com/cncf/cnf-wg/pull/134

File: charter.md (hunk 1 above)
Comment (taylor) [reply charter.md:20]: These changes have been moved to the new PR https://github.com/cncf/cnf-wg/pull/134

Comment (taylor) [reply charter.md:20]: These changes are now in a new PR https://github.com/cncf/cnf-wg/pull/134

File: charter.md (hunk 2 above)
Comment (taylor) [reply charter.md:21]: These changes are now in the new PR https://github.com/cncf/cnf-wg/pull/134

**Review by taylor — APPROVED** [review]
lgtm.  new updates will come in a new PR. eg. https://github.com/cncf/cnf-wg/pull/134

**Review by jeffsaelens — APPROVED** [review]
I'm good to push this through, and then continue to drive through with the rework in the other PR.

**Review by Smitholi67 — APPROVED** [review]
lgtm

**Review by rabi-abdel — APPROVED** [review]

**Review by xmulligan — APPROVED** [review]

**Review by ildikov — APPROVED** [review]

**Review by electrocucaracha — APPROVED** [review]
